### ISSUE TITLE
Plugin Causes App to Crash on Android: "Error connecting to the service protocol"

### ISSUE THREAD

**Issue by ekleel** [pichillilorenzo/flutter_inappwebview#267]
## Environment

**Flutter version**: Channel stable, v1.12.13+hotfix.5
**Plugin version**: 2.1.0+1
**Android version:**  Android 9 & 10
**Device information:**    
- Actual Pixel 2 (Android 10)
- Pixel 3 XL (Android 9) Emulator
- Pixel 3 XL (Android 10) Emulator

## Description

I'm getting this error on a fresh install of Flutter:
```
Error connecting to the service protocol: failed to connect to
http://127.0.0.1:63275/aSiGzQCEBpU=/
```
I got this error the first time on an actual project that I was working on. So I tried a fresh Flutter project to see if the issue was caused by `flutter_inappwebview`. The test app runs without issues before installing `flutter_inappwebview` but crashes right after installing it. I've tried `flutter clean` and `flutter run` again but no results. I've also tried an actual device and two emulators with Android 9 and Android 10 but crashes on all of them citing the same message.

## Steps to reproduce

1. Create a new Flutter app
2. Install flutter_inappwebview
3. Run on Android (device or emulator on Android 9 or 10)

## Stacktrace/Logcat  
Terminal Logs:
```
flutter run
Using hardware rendering with device Android SDK built for
x86. If you get graphics artifacts, consider enabling
software rendering with "--enable-software-rendering".
Launching lib/main.dart on Android SDK built for x86 in
debug mode...
Running Gradle task 'assembleDebug'...                                                                                   Note: Some input files use or override a deprecated API.
Running Gradle task 'assembleDebug'...                                                                                      Note: Recompile with -Xlint:deprecation for details.
Running Gradle task 'assembleDebug'...                                                                                      Note: Some input files use unchecked or unsafe operations.
Running Gradle task 'assembleDebug'...                                                                                      Note: Recompile with -Xlint:unchecked for details.
Running Gradle task 'assembleDebug'...                                  
Running Gradle task 'assembleDebug'... Done                        38.6s
✓ Built build/app/outputs/apk/debug/app-debug.apk.
Installing build/app/outputs/apk/app.apk...                         5.8s
Error connecting to the service protocol: failed to connect to
http://127.0.0.1:63275/aSiGzQCEBpU=/
```
Flutter Doctor:
```
[✓] Flutter (Channel stable, v1.12.13+hotfix.5, on Mac OS X 10.14.4 18E226, locale en-SA)
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
[✓] Android Studio (version 3.5)
[✓] VS Code (version 1.41.1)
[✓] Connected device (1 available)
• No issues found!
```
Widget:
```
Widget build(BuildContext context) {
  return Scaffold(
    appBar: AppBar(
      title: Text(widget.title),
    ),
    body: Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: <Widget>[
          InAppWebView(
            initialUrl: "https://flutter.dev/",
            initialHeaders: {},
            initialOptions: InAppWebViewWidgetOptions(
              inAppWebViewOptions: InAppWebViewOptions(
                debuggingEnabled: true,
              ),
            ),
          ),
        ],
      ),
    ),
  );
}
```
--
Any ideas how to solve this?

**Comment by ekleel** [comment]
More Logs with `flutter run -v`:

```
emulator-5554 shell am start -a android.intent.action.RUN
-f 0x20000000 --ez enable-background-compilation true
--ez enable-dart-profiling true --ez enable-checked-mode
true --ez verify-entry-points true
com.example.test_android_ptcl/com.example.test_android_pt
cl.MainActivity
[ +236 ms] Starting: Intent {
act=android.intent.action.RUN flg=0x20000000
cmp=com.example.test_android_ptcl/.MainActivity (has
extras) }
[   +1 ms] Waiting for observatory port to be
available...
[+3568 ms] Observatory URL on device:
http://127.0.0.1:40565/xRfgRCC7WFU=/
[   +3 ms] executing:
/Users/MacBook/Library/Android/sdk/platform-tools/adb -s
emulator-5554 forward tcp:0 tcp:40565
[  +18 ms] 54109
[        ] Forwarded host port 54109 to device port 40565
for Observatory
[  +26 ms] Connecting to service protocol:
http://127.0.0.1:54109/xRfgRCC7WFU=/
[ +155 ms] Fail to connect to service protocol:
http://127.0.0.1:54109/xRfgRCC7WFU=/: HttpException:
Connection closed before full header was received, uri =
http://127.0.0.1:54109/xRfgRCC7WFU=/ws
[   +6 ms] Error connecting to the service protocol:
failed to
           connect to
http://127.0.0.1:54109/xRfgRCC7WFU=/
[  +15 ms] "flutter run" took 45,936ms.

#0      throwToolExit
(package:flutter_tools/src/base/common.dart:28:3)
#1      RunCommand.runCommand
(package:flutter_tools/src/commands/run.dart:509:7)
<asynchronous suspension>
#2      FlutterCommand.verifyThenRunCommand
(package:flutter_tools/src/runner/flutter_command.dart:61
5:18)
<asynchronous suspension>
#3      FlutterCommand.run.<anonymous closure>
(package:flutter_tools/src/runner/flutter_command.dart:51
5:33)
<asynchronous suspension>
#4      AppContext.run.<anonymous closure>
(package:flutter_tools/src/base/context.dart:146:29)
#5      _rootRun (dart:async/zone.dart:1126:13)
#6      _CustomZone.run (dart:async/zone.dart:1023:19)
#7      _runZoned (dart:async/zone.dart:1518:10)
#8      runZoned (dart:async/zone.dart:1465:12)
#9      AppContext.run
(package:flutter_tools/src/base/context.dart:145:18)
#10     FlutterCommand.run
(package:flutter_tools/src/runner/flutter_command.dart:50
5:20)
#11     CommandRunner.runCommand
(package:args/command_runner.dart:197:27)
#12     FlutterCommandRunner.runCommand.<anonymous
closure>
(package:flutter_tools/src/runner/flutter_command_runner.
dart:413:21)
<asynchronous suspension>
#13     AppContext.run.<anonymous closure>
(package:flutter_tools/src/base/context.dart:146:29)
#14     _rootRun (dart:async/zone.dart:1126:13)
#15     _CustomZone.run (dart:async/zone.dart:1023:19)
#16     _runZoned (dart:async/zone.dart:1518:10)
#17     runZoned (dart:async/zone.dart:1465:12)
#18     AppContext.run
(package:flutter_tools/src/base/context.dart:145:18)
#19     FlutterCommandRunner.runCommand
(package:flutter_tools/src/runner/flutter_command_runner.
dart:364:19)
#20     CommandRunner.run.<anonymous closure>
(package:args/command_runner.dart:112:25)
#21     new Future.sync (dart:async/future.dart:224:31)
#22     CommandRunner.run
(package:args/command_runner.dart:112:14)
#23     FlutterCommandRunner.run
(package:flutter_tools/src/runner/flutter_command_runner.
dart:250:18)
#24     run.<anonymous closure>.<anonymous closure>
(package:flutter_tools/runner.dart:63:22)
#25     _rootRun (dart:async/zone.dart:1126:13)
#26     _CustomZone.run (dart:async/zone.dart:1023:19)
#27     _runZoned (dart:async/zone.dart:1518:10)
#28     runZoned (dart:async/zone.dart:1502:12)
#29     run.<anonymous closure>
(package:flutter_tools/runner.dart:61:18)
<asynchronous suspension>
#30     AppContext.run.<anonymous closure>
(package:flutter_tools/src/base/context.dart:146:29)
#31     _rootRun (dart:async/zone.dart:1126:13)
#32     _CustomZone.run (dart:async/zone.dart:1023:19)
#33     _runZoned (dart:async/zone.dart:1518:10)
#34     runZoned (dart:async/zone.dart:1465:12)
#35     AppContext.run
(package:flutter_tools/src/base/context.dart:145:18)
#36     runInContext
(package:flutter_tools/src/context_runner.dart:64:24)
#37     run (package:flutter_tools/runner.dart:50:10)
#38     main (package:flutter_tools/executable.dart:65:9)
#39     main
(file:///Users/MacBook/Sites/flutter/flutter/packages/flu
tter_tools/bin/flutter_tools.dart:8:3)
#40     _startIsolate.<anonymous closure>
(dart:isolate-patch/isolate_patch.dart:305:32)
#41     _RawReceivePortImpl._handleMessage
(dart:isolate-patch/isolate_patch.dart:174:12)
```

**Comment by tuanha2000vn** [comment]
I have the same issue.

Strange enough, my old app use the exactly   flutter_inappwebview: ^2.1.0+1 and not crashing at all....

If you know how to fix it, please let me know.

Thank you.

**Comment by ekleel** [comment]
Sorry, not yet. I'm still facing this issue.

**Comment by tuanha2000vn** [comment]
@ekleel I think there's something wrong on the pub.dev package, just pull it directly from github and it work.
`  flutter_inappwebview:
    git:
      url: https://github.com/pichillilorenzo/flutter_inappwebview.git`

**Comment by bpbunthoeun** [comment]
> @ekleel I think there's something wrong on the pub.dev package, just pull it directly from github and it work.
> 
> ```
>   flutter_inappwebview:
>     git:
>       url: https://github.com/pichillilorenzo/flutter_inappwebview.git
> ```

Yes, I got the same problem and I follow you now it is working as well. thanks

**Comment by Willian199** [comment]
Same problem,

> I have the same issue.
> 
> Strange enough, my old app use the exactly flutter_inappwebview: ^2.1.0+1 and not crashing at all....
> 
> If you know how to fix it, please let me know.
> 
> Thank you.

but this works

**Comment by ekleel** [comment]
> @ekleel I think there's something wrong on the pub.dev package, just pull it directly from github and it work.
> 
> ```
>   flutter_inappwebview:
>     git:
>       url: https://github.com/pichillilorenzo/flutter_inappwebview.git
> ```

I can confirm, this works. Thank you!

**Comment by Faiyyaz** [comment]
> @ekleel I think there's something wrong on the pub.dev package, just pull it directly from github and it work.
> 
> ```
>   flutter_inappwebview:
>     git:
>       url: https://github.com/pichillilorenzo/flutter_inappwebview.git
> ```

I can confirm, this works. Thank you!
But can we get it merge officially and get a updated version

**Comment by pichillilorenzo** [comment]
Released new version `3.0.0`!
